documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -3,90 +3,126 @@
-Grants are cached and enforced locally on the Kafka broker for each user session to provide fast authorization decisions. Because they are refreshed, any changes to the grants on the Keycloak server are detected and enforced.
+Grants are cached and enforced locally on the Kafka broker for each user session to provide fast authorization decisions. Because they are periodically refreshed, any changes to the grants on the Keycloak server are detected and enforced.
 
+== Provisioning the Keycloak server
 
-== Starting the pods
+The easiest way to install the Keycloak server is to use the Keycloak Operator. Follow the instructions in {keycloak-server-install-doc}.
 
-For this example, we use the kubernetes deployment scripts available at {oauth-kubernetes-examples-link}:
+== Uploading the demo realm
 
-[source,yaml,subs="attributes"]
+Once a Keycloak instance is running, we can upload the example realm by using the Keycloak Admin Console
+
+The following instructions can be used regardless of how your Keycloak instance was provisioned. If Keycloak Operator was used, you can follow the steps to obtain the necessary information to continue, otherwise you may have to discover the necessary information for the next steps yourself.
+
+First, determine the external hostname to access the Admin Console.
+
+For example, if the Keycloak Operator was used to install Keycloak server in namespace called `sso`, you can run:
+
+[source,shell,subs="attributes"]
 ----
-export ROOT=https://raw.githubusercontent.com/strimzi/strimzi-kafka-oauth/{oauth-artifact-version}
+NS=sso
+kubectl get ingress keycloak -n $NS
 ----
 
-.Deploy the Postgres database for Keycloak
+Second, determine the password for the admin user.
+
+If you used Keycloak Operator, the password is stored as a secret, so you first have to identify the name of the secret. You can inspect the yaml for the Keycloak instance. For example:
 
-[source,yaml]
+[source,shell,subs="attributes"]
 ----
-kubectl apply -f $ROOT/examples/kubernetes/postgres-pvc.yaml
-kubectl apply -f $ROOT/examples/kubernetes/postgres.yaml
+kubectl get -n $NS pod keycloak-0 -o yaml | less
 ----
 
-.Deploy the Keycloak server
+Search for `KEYCLOAK_PASSWORD`, and note the `secretKeyRef`.`name`. It is usually something like `credential-keycloak`.
+
+You can then obtain the clear text password by running:
 
-[source,yaml]
+[source,shell,subs="attributes"]
 ----
-kubectl apply -f $ROOT/examples/kubernetes/keycloak-realms-configmap.yaml
-kubectl apply -f $ROOT/examples/kubernetes/keycloak-postgres.yaml
+SECRET_NAME=credential-keycloak
+kubectl get -n $NS secret $SECRET_NAME -o yaml | grep PASSWORD | awk '{print $2}' | base64 -D
 ----
 
-If your default namespace is not `myproject` (for example, `default`), use the following to deploy Keycloak:
+Now, login to the Admin Console (you have to use https://HOST for Kubernetes ingress) using `admin` as the username, and the revealed password.
+
```

Comment:
I think line 71 should be two separate steps.

Reply:
@laidan6000 Can you be more specific which line? Line 71 seems to be an empty line.

Reply:
Sorry, line 69.

When you are logged in, click *Add Realm* to import the example realm.

Select the `$EXAMPLES_DIR/security/keycloak-authorization/kafka-authz-realm.json` file from your disk, and then click *Create*.